File: Website SEO/public_html/check_js.py
import os
import glob
import subprocess
import re

html_files = glob.glob('/Users/nahid/Desktop/Nahid/AI Profit Lab/Website/Website SEO/public_html/**/*.html', recursive=True)
has_errors = False

for file in html_files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()

    scripts = re.findall(r'<script\b[^>]*>([\s\S]*?)<\/script>', content, re.IGNORECASE)
    
    for idx, script in enumerate(scripts):
        if not script.strip(): continue
        
        with open('tmp_script.js', 'w', encoding='utf-8') as f:
            f.write(script)
            
        res = subprocess.run(['node', '-c', 'tmp_script.js'], capture_output=True, text=True)
        if res.returncode != 0:
            if "await is only valid in async functions" in res.stderr:
                with open('tmp_script.js', 'w', encoding='utf-8') as f:
                    f.write('(async () => {\n' + script + '\n})();')
                res2 = subprocess.run(['node', '-c', 'tmp_script.js'], capture_output=True, text=True)
                if res2.returncode != 0:
                    print(f"Syntax Error in {file} script #{idx + 1}")
                    print(res2.stderr.splitlines()[0])
                    has_errors = True
            elif "Return statement is not allowed here" in res.stderr:
                with open('tmp_script.js', 'w', encoding='utf-8') as f:
                    f.write('function dummy() {\n' + script + '\n}')
                res2 = subprocess.run(['node', '-c', 'tmp_script.js'], capture_output=True, text=True)
                if res2.returncode != 0:
                    print(f"Syntax Error in {file} script #{idx + 1}")
                    has_errors = True
            else:
                print(f"Syntax Error in {file} script #{idx + 1}")
                print(res.stderr.splitlines()[0])
                has_errors = True

if not has_errors:
    print("All scripts passed syntax check!")

File: Website SEO/public_html/check_js.js
const fs = require('fs');
const path = require('path');
const glob = require('glob');

// Basic regex to extract script tags
const extractScripts = (html) => {
    const scripts = [];
    const regex = /<script\b[^>]*>([\s\S]*?)<\/script>/gi;
    let match;
    while ((match = regex.exec(html)) !== null) {
        if (match[1].trim() !== '') {
            scripts.push({
                content: match[1],
                start: match.index
            });
        }
    }
    return scripts;
};

// We can just use acorn or basic eval to check syntax
// Since acorn isn't definitely installed, we can just spawn a node process with the script content to check syntax
const { execSync } = require('child_process');

const files = glob.sync('/Users/nahid/Desktop/Nahid/AI Profit Lab/Website/Website SEO/public_html/**/*.html');
let hasErrors = false;

files.forEach(file => {
    const content = fs.readFileSync(file, 'utf8');
    const scripts = extractScripts(content);
    
    scripts.forEach((script, idx) => {
        const tmpFile = path.join(__dirname, 'tmp_script.js');
        // Wrap in an async function if it uses await outside
        let jsCode = script.content;
        fs.writeFileSync(tmpFile, jsCode);
        try {
            execSync(`node -c tmp_script.js`, { stdio: 'ignore' });
        } catch (e) {
            // Check if error is just about 'await' being a reserved word outside async
            if (e.message && e.message.includes("await")) {
                fs.writeFileSync(tmpFile, `(async () => { ${jsCode} })();`);
                try {
                     execSync(`node -c tmp_script.js`, { stdio: 'ignore' });
                } catch (e2) {
                     console.log(`Syntax Error in ${file} script #${idx + 1}`);
                     hasErrors = true;
                }
            } else {
                console.log(`Syntax Error in ${file} script #${idx + 1}`);
                hasErrors = true;
            }
        }
    });
});
if (!hasErrors) console.log("All scripts passed syntax check!");
